### ISSUE TITLE
Add a `disable_peers` field to `testnet::Parameters`

### ISSUE THREAD

**Issue by arya2** [ZcashFoundation/zebra#8361]
## Motivation

We want to disable non-local peer connections for `Regtest` and configured Testnets with a `disable_peers` field on `Parameters`.

### Note

Make sure to search for the issue number and address any TODOs.

**Comment by mpguerra** [comment]
@arya2 is this still necessary?

**Comment by arya2** [comment]
> @arya2 is this still necessary?

No, returning an empty set from the `zebra_network::Config::initial_peers()` method on Regtest is fine, at least for now.

It could be useful for creating isolated custom Testnets later, which would be like custom Regtest networks (Zebra's Regtest currently tries to closely match the zcashd Regtest parameters, users may want to closely match the Mainnet parameters instead).

**Comment by arya2** [comment]
This issue is still low-priority cleanup so that it's easy to configure Zebra to disallow peers on a custom Testnet, but it turned out zcashd's Regtest allows local connections and it's likely easier to fix this issue in parallel.

**Comment by mpguerra** [comment]
not a priority right now
